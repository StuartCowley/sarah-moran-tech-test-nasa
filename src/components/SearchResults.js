import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ searched, results }) => {
  if (!searched) return null;
  if (!results.length) {
    return <p className="no-results">No Results</p>;
  } else {
    return (
      <>
        <div className="image-gallery">
          {results.map((image) => (
            <img
              className="card-image"
              src={image}
              alt="requested-result"
              key={image}
            />
          ))}
        </div>
      </>
    );
  }
};

export default SearchResults;
