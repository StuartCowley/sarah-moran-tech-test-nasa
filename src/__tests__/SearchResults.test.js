import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders results", () => {
    const results = ["image1", "image2", "image3"];
    const { asFragment } = render(<SearchResults results={results} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders images if there are results", () => {
    const results = ["image1", "image2", "image3"];
    const searched = true;
    render(<SearchResults results={results} searched={searched} />);
    const imageResults = screen.queryAllByAltText("requested-result");

    expect(imageResults.length).toBe(3);
  });

  it("renders no results text if there are no results", () => {
    render(<SearchResults results={[]} searched={true} />);
    const noResults = screen.getByText("No Results");

    expect(noResults).toBeInTheDocument();
  });
});
