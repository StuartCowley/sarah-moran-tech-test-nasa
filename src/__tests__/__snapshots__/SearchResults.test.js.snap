// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `<DocumentFragment />`;

exports[`SearchResults renders results 1`] = `<DocumentFragment />`;
